### ISSUE TITLE
[ElementWise]: operation PROD has incompatible input types Float and Bool

### ISSUE THREAD

**Issue by RXWein** [NVIDIA/TensorRT#1872]
## Description
Hi, I has generate the grid sample by using the plugin and want to use elementwise operation to filter the invalid element in the grid. This is realized by using a mask of bool type to get the product of valid element. The code snippet is:
voxel = grid_sample(network, input_tensor, grid, logger)
voxel = network.add_elementwise(voxel.get_output(0), valid_mask.get_output(0), trt.ElementWiseOperation.PROD)

But it throws error
[TensorRT] ERROR: 4: [layers.cpp::validate::2292] Error Code 4: Internal Error ((Unnamed Layer* 728) [ElementWise]: operation PROD has incompatible input types Float and Bool)

I also tried to convert the mask of bool type to float before doing the elementwise operation by using set_output_type(0, trt.DataType.FLOAT), and it also throws the same error. It's strange that it seems the set_output_type doesn't work.

## Environment

**TensorRT Version**: 8.0
**NVIDIA GPU**: Agx xavier
**NVIDIA Driver Version**: 
**CUDA Version**: 10.2
**CUDNN Version**: 8.2
**Operating System**: Linux
**Python Version (if applicable)**: 3.6
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if so, version)**:

**Comment by samurdhikaru** [comment]
@RXWein You can use an Identity layer to do a cast from `trt.DataType.BOOL` to ` trt.DataType.FLOAT`. 

```
valid_mask_bool = network.add_identity(valid_mask.get_output(0))
valid_mask_bool .set_output_type(0, trt.DataType.FLOAT)
```

Then you can pass `valid_mask_bool` to  `add_elementwise`.

**Comment by RXWein** [comment]
@samurdhikaru Thank you, that works for me
